src/App.tsx

```diff
@@ -1,27 +1,91 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
+import { Box, Button, Typography } from '@mui/material';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good []>([]);
+  const [activeButton, setActiveButton] = useState('');
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const loadAllGoods = useCallback(async () => {
+    const allGoods = getAll();
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    setGoods(await allGoods);
+    setActiveButton('load all goods');
+  }, []);
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  const load5First = useCallback(async () => {
+    const fiveFirst = get5First();
 
-    <GoodsList goods={[]} />
-  </div>
-);
+    setGoods(await fiveFirst);
+    setActiveButton('load 5 first goods');
+  }, []);
+
+  const loadRedGoods = useCallback(async () => {
+    const redGoods = getRedGoods();
+
+    setGoods(await redGoods);
+    setActiveButton('load red goods');
+  }, []);
```

Comment:
Maybe, let's create one function that will handle all cases ?

Reply:
good idea! added